### PR TITLE
Use prebuilt macOS environment on Travis CI

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ jobs:
       compiler: gcc
       cache: ccache
       # Ubuntu Bionic build prerequisites
-      env: CMAKEFLAGS_EXTRA="-DLOCALECOMPARE=ON"
+      env: CMAKEFLAGS_EXTRA="-DLOCALECOMPARE=ON -DFAAD=ON -DWAVPACK=ON -DMAD=ON -DMODPLUG=ON"
```
Comment (Be-ing) [inline .travis.yml:73]: Why did you change the Ubuntu build flags? I thought we were using FFMPEG now, not FAAD?

File: .travis.yml [hunk 2]
```diff
@@ -139,19 +139,28 @@ jobs:
...
       # instead of <opus/opus_multistream.h>.
       # Virtual X (Xvfb) is needed for analyzer waveform tests
+      addons:
+        homebrew:
+          update: true
+          packages:
+            - ccache
       env: >-
-        CMAKEFLAGS_EXTRA="-DLOCALECOMPARE=OFF -DCMAKE_PREFIX_PATH=/usr/local/opt/qt5/"
+        MIXXX_ENVNAME=2.3-j00004-497fe02e-osx10.11-x86_64-release
```
Comment (Be-ing) [inline .travis.yml:148]: Can you get Travis to read the build/osx/golden_environment file? If we have to update that in multiple places, we will forget to update it everywhere.

File: .travis.yml [hunk 3]
```diff
@@ -139,19 +139,28 @@ jobs:
...
+        MIXXX_ENVPATH="${HOME}/buildenv"
+        CMAKEFLAGS_EXTRA="-DLOCALECOMPARE=OFF -DFAAD=OFF -DWAVPACK=OFF -DWAVPACK=OFF -DMAD=OFF -DMODPLUG=OFF -DCMAKE_PREFIX_PATH=${MIXXX_ENVPATH} -DQt5_DIR=${MIXXX_ENVPATH}/Qt-5.12.3/lib/cmake/Qt5/"
```
Comment (Be-ing) [inline .travis.yml:150]: Do we have to hardcode the Qt version here? Would `*` work?
```suggestion
        CMAKEFLAGS_EXTRA="-DLOCALECOMPARE=OFF -DFAAD=OFF -DWAVPACK=OFF -DWAVPACK=OFF -DMAD=OFF -DMODPLUG=OFF -DCMAKE_PREFIX_PATH=${MIXXX_ENVPATH} -DQt5_DIR=${MIXXX_ENVPATH}/Qt-*/lib/cmake/Qt5/"
```

File: .travis.yml (hunk 1 above)
Comment (Holzhaus) [reply .travis.yml:73]: These weren't actually changed: https://github.com/mixxxdj/mixxx/pull/2791/files#diff-354f30a63fb0907d4ad57269548329e3R20

File: .travis.yml (hunk 3 above)
Comment (Holzhaus) [reply .travis.yml:150]: That was my initial attempt but didn't work for some reason. I can try again.

File: .travis.yml (hunk 2 above)
Comment (Holzhaus) [reply .travis.yml:148]: What file? I can't find it... https://downloads.mixxx.org/builds/buildserver/2.3.x-macosx/

Comment (Be-ing) [reply .travis.yml:148]: It's in this Git repository.

File: .travis.yml (hunk 3 above)
Comment (Holzhaus) [reply .travis.yml]: Does not work unfortunately: https://travis-ci.org/github/Holzhaus/mixxx/jobs/687880337#L14832

Comment (Be-ing) [reply .travis.yml]: How do you know the expression evaluated correctly? It is not echoed anywhere in the log.

Comment (Holzhaus) [reply .travis.yml]: If you take the current state of this PR and replace the Qt version with a glob it stops working.

**Comment by Holzhaus** [comment]
No idea where the linker error comes from. Any ideas?

**Comment by Holzhaus** [comment]
> No idea where the linker error comes from. Any ideas?

The linker error is because the taglib-config file hardcodes a specific path:
```
$ less lib/pkgconfig/taglib.pc
prefix=/Users/mixxx/bs-2.3-mac/amd64/environment/2.3-j00004-497fe02e-osx10.11-x86_64-release
exec_prefix=/Users/mixxx/bs-2.3-mac/amd64/environment/2.3-j00004-497fe02e-osx10.11-x86_64-release
libdir=/Users/mixxx/bs-2.3-mac/amd64/environment/2.3-j00004-497fe02e-osx10.11-x86_64-release/lib
includedir=/Users/mixxx/bs-2.3-mac/amd64/environment/2.3-j00004-497fe02e-osx10.11-x86_64-release/include

Name: TagLib
Description: Audio meta-data library
Requires:
Version: 1.11.1
Libs: -L${libdir} -ltag
Cflags: -I${includedir}/taglib
```

**Comment by Holzhaus** [comment]
Ok, I worked around the broken buildenv by replacing the paths. It now builds fine, but the tests do not work because it doesn't find the Qt platform plugin:

    qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""
    This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

I don't know enough about macOS to fix this. Ideas?

**Comment by uklotzde** [comment]
Hooray! I tried something similar to fix the tests, but then loading of the sqlite drivers still failed. Maybe I didn't add it to the global environment.

**Comment by uklotzde** [comment]
LGTM @Be Ready?

**Comment by Be-ing** [comment]
Thanks for working on this! Should we merge it to the 2.3 branch?

**Comment by Holzhaus** [comment]
Okay, I retargeted this PR to 2.3.
